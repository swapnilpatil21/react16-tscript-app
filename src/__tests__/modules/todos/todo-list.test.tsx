import { shallow } from "enzyme";
import TodoList, { ITodoList } from "../../../modules/todos/components/todo-list";

describe("<TodoList />", ()=> {
  const defaultprops: ITodoList = {
    deleteTodoHandler() {
      return;
    },
    todos: [{id: 1, text: "text one"}, {id: 2, text: "text two"}],
  }

  it("should render correctly", () => {
    const wrapperComponent = shallow(<TodoList {...defaultprops} />)

    expect(wrapperComponent.find('p b').text()).toEqual("TodoList")
    expect(wrapperComponent.find('li')).toHaveLength(2)
    expect(wrapperComponent).toMatchSnapshot()
  })

  it("should render with empty list", () => {
    const emptyTodo = {
      todos: []
    }
    const wrapperComponent = shallow(<TodoList {...defaultprops} {...emptyTodo} />)

    expect(wrapperComponent.find('p b').text()).toEqual("TodoList")
    expect(wrapperComponent.find('ul').text()).toEqual("You don't have todo items")
    expect(wrapperComponent.find('li')).toHaveLength(0)
    expect(wrapperComponent).toMatchSnapshot()
  })

  it("should called delete buttom", () => {
    const deleteTodoHandler = jest.fn();
    const wrapperComponent = shallow(<TodoList {...defaultprops} deleteTodoHandler={deleteTodoHandler} />)

    wrapperComponent.find('button').first().simulate('click')    
    expect(deleteTodoHandler).toBeCalled()
  })
})