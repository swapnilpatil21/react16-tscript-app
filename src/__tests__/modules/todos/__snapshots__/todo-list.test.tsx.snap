// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> should render correctly 1`] = `
<div>
  <p>
    <b>
      TodoList
    </b>
  </p>
  <ul>
    <li
      key="1"
    >
      text one
       
      <span>
        <button
          data-testid="Delete"
          onClick={[Function]}
        >
          Delete
        </button>
      </span>
    </li>
    <li
      key="2"
    >
      text two
       
      <span>
        <button
          data-testid="Delete"
          onClick={[Function]}
        >
          Delete
        </button>
      </span>
    </li>
  </ul>
</div>
`;

exports[`<TodoList /> should render with empty list 1`] = `
<div>
  <p>
    <b>
      TodoList
    </b>
  </p>
  <ul>
    You don't have todo items
  </ul>
</div>
`;
